### PR TITLE
mc-consensus-mint-client improvements for offline/HSM flows

### PR DESCRIPTION
**Pull request by eranrund**
`mc-consensus-mint-client` is used to generate and sign `MintConfigTx`s and `MintTx`s. Prior to this PR, this requires having the private key(s) used for signing inside PEM file that is readable by the utility. This PR enables a new flow where the data to be signed (the hash of the tx prefix) can be generated, and then that hash can be signed offline or by an HSM. The produced signature, in hex bytes, can be fed back into the utility to generate a transaction.

Example usage:
1) Generate a hash for a `MintConfigTx`: `mc-consensus-mint-client hash-mint-config-tx --token-id 10 --config 10:1:minting1-public.pem --total-mint-limit 40 --tombstone 123 --nonce 33333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333`
2) Take the generated hash (presented as hex-encoded bytes) and sign it with an HSM
3) Take the signature (as hex-encoded bytes) and use it to generate and submit a `MintConfigTx`: `mc-consensus-mint-client generate-and-submit-mint-config-tx --node mc://node1.eran.mobilecoin.com/ --token-id 10 --config 10:1:minting1-public.pem --total-mint-limit 40 --tombstone 123 --nonce 33333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333 --signature 263f5c3da1f1e6d988ab2fb9c366e21f8a218d6c88c85add50e417d93dc9ef8f66ac4099aeea5b69adb2685569e34ffd07b7ab5b092a06d8901aec663ed3360e`

### CODE REVIEW

File: consensus/mint-client/src/config.rs [hunk 1]
```diff
@@ -247,6 +333,14 @@ pub fn load_key_from_pem(filename: &str) -> Result<Ed25519Private, String> {
...
 }
 
+pub fn parse_ed25519_signature(hex_signature: &str) -> Result<Ed25519Signature, String> {
+    let bytes = hex::decode(hex_signature)
+        .map_err(|err| format!("Failed decoding hex signature: {}", err))?;
+
+    Ed25519Signature::try_from(&bytes[..])
+        .map_err(|err| format!("Failed parsing Ed25519 signature: {}", err))
+}
+
```
Comment (jcape) [inline consensus/mint-client/src/config.rs:343]: Do we want to use bytes here, or PEM/DER: https://github.com/mobilecoinfoundation/mobilecoin/blob/9450b1966cef23cedd53cac5f169110019fe2f2d/crypto/keys/src/ed25519.rs#L78

**Review by jcape — APPROVED** [review]
Looks reasonable, but we may want to output PEM for the signatures instead, depending on what the wallets are going to spit out.

File: consensus/mint-client/src/config.rs (hunk 1 above)
Comment (eranrund) [reply consensus/mint-client/src/config.rs:343]: I have no preference, and this is easy to change. @holtzman might have an opinion, as he's been playing with HSMs.

Comment (holtzman) [reply consensus/mint-client/src/config.rs:343]: Trezor's APIs are in Python, and it's easier to deal with it in hex or base64 rather than PEM.

Comment (eranrund) [reply consensus/mint-client/src/config.rs:343]: @jcape are you okay leaving it as bytes? or do you want me to switch to PEM+DER?

**Review by garbageslam — APPROVED** [review]
LGTM

(I have no particular thoughts on the hex vs. base64 vs. der encoded)

File: consensus/mint-client/src/config.rs (hunk 1 above)
Comment (jcape) [reply consensus/mint-client/src/config.rs:343]: If the wallets is using hex, and the signature component order---Ed25519 signatures have two components---matches that's fine.

I expect "industrial" HSMs to do PEM, which will require some massaging to get into the correct hex format.

**Comment by eranrund** [comment]
@jcape following our conversation yesterday I modified this to support both hex-encoded bytes or a der/pem filename.

**Pull request merged**